### PR TITLE
Bump go version to 1.22, bump linter

### PR DESCRIPTION
**Pull request by haydentherapper**

### CODE REVIEW

**Comment by haydentherapper** [comment]
For some reason, codeql is downloading 2.16.3 rather than 2.16.4 which includes go 1.22. will hold off for now

**Review by cpanato — DISMISSED** [review]
besides the codeql
lgtm

**Comment by haydentherapper** [comment]
Nothing to do at this point, codeql is being pulled from the hosted tool cache, so we just need to wait for that to be updated.

**Comment by haydentherapper** [comment]
Toolchain now contains 2.16.4 but Go 1.22 support is still missing even though the docs say it should be present. will wait til 2.16.5 i guess

**Comment by haydentherapper** [comment]
@cpanato finally figured it out, I sent the go.mod go version to the specific version of Go because apparently go 1.22 is a development version. hopefully this is correct now

File: .github/workflows/verify.yml [hunk 1]
```diff
@@ -50,5 +50,5 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@3cfe3a4abbb849e10058ce4af15d205b6da42804 # v3
         with:
-          version: v1.54
+          version: v1.56
```
Comment (cpanato) [inline .github/workflows/verify.yml:53]: can you update to `v1.57` ?

File: .github/workflows/verify.yml [hunk 2]
```diff
@@ -50,5 +50,5 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@3cfe3a4abbb849e10058ce4af15d205b6da42804 # v3
```
Comment (cpanato) [inline .github/workflows/verify.yml:51]: as you are here can you update the version comment to the full semver as well?

**Review by cpanato — CHANGES_REQUESTED** [review]
thanks! some small cr

File: .github/workflows/verify.yml (hunk 1 above)
Comment (haydentherapper) [reply .github/workflows/verify.yml]: done

Comment (cpanato) [reply .github/workflows/verify.yml]: thanks!

**Review by cpanato — APPROVED** [review]

**Pull request merged**